### PR TITLE
Add a case to test ORC writing/reading with lots of nulls

### LINKED ISSUE #8731: ORC reads at scale with all null values: Like `OrcQuerySuite.scala#L173`, but with large number of rows.
Add test case according to [Spark test case](https://github.com/apache/spark/blob/v3.4.0/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala#L173)

```
  test("save and load case class RDD with `None`s as orc") {
    val data = (
      Option.empty[Int],
      Option.empty[Long],
      Option.empty[Float],
      Option.empty[Double],
      Option.empty[Boolean]
    ) :: Nil

    withOrcFile(data) { file =>
      checkAnswer(
        spark.read.orc(file),
        Row(Seq.fill(5)(null): _*))
    }
  }
```

### PR DESCRIPTION
**Pull request by res-life**
closes #8731 

This is to test large number of nulls.
- Add a case to test ORC writing/reading with lots of nulls.
- Add large_data_test mark in pytest, by default, disable the tests with large_data_test mark.

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
This case can pass, but causes this a new issue: https://github.com/NVIDIA/spark-rapids/issues/8826.
I suggest merge this PR and investigate the new issue

**Review by revans2 — APPROVED** [review]

File: integration_tests/src/main/python/orc_test.py [hunk 1]
```diff
@@ -902,3 +902,32 @@ def gen_null_df(spark):
...
+        lambda spark, path: gen_null_df(spark).write.orc(path),
+        lambda spark, path: spark.read.orc(path),
+        data_path,
+        conf=all_confs)
+    gpu_file_path = data_path + "/CPU"
+    sqls = ["SELECT * FROM my_large_table",
+            "SELECT * FROM my_large_table WHERE c2 = 5",
+            "SELECT COUNT(*) FROM my_large_table WHERE c3 IS NOT NULL",
+            "SELECT * FROM my_large_table WHERE c4 IS NULL",
+            "SELECT * FROM my_large_table WHERE c5 IS NULL",
```
Comment (mythrocks) [inline integration_tests/src/main/python/orc_test.py:927]: :+1:

**Review by mythrocks — APPROVED** [review]
LGTM. Thank you! This covers the cases I was hoping for.

Per @revans2's advice, we can move this under the scale tests as a follow-on.

File: integration_tests/README.md [hunk 2]
```diff
@@ -398,6 +398,10 @@ properly without it. These tests assume Delta Lake is not configured and are dis
 If Spark has been configured to support Delta Lake then these tests can be enabled by adding the
 `--delta_lake` option to the command.
 
+### Enabling large data tests
```
[the remark this answers is not in the document]
Comment (res-life) [reply integration_tests/README.md:401]: Yes, it's not running. As discussed with Tim and Gary before, will create another CI job to run this kind of tests using:
```
 -m large_date_test --large_date_test
```
It's fine to move to scale test. The follow-up issue is: https://github.com/NVIDIA/spark-rapids/issues/8849

**Pull request merged**